src/index.tsx

```diff
@@ -12,16 +14,21 @@ declare module '@tanstack/react-router' {
   interface Register {
     router: typeof router;
   }
-}
+};
+
+const queryClient = new QueryClient();
 
 const rootElement = document.getElementById('app')!;
 
 if (!rootElement.innerHTML) {
   ReactDOM.createRoot(rootElement).render(
     <StrictMode>
-      <MarigoldProvider theme={theme}>
-        <RouterProvider router={router} />
-      </MarigoldProvider>
+      <QueryClientProvider client={queryClient}>
+        <MarigoldProvider theme={theme}>
+          <RouterProvider router={router} />
+          <ReactQueryDevtools />
```

Comment:
I would add this line to __root.tsx next to <TanStackRouterDevtools />.  Then it's all together. 


It may also not be ideal in root.tsx and we should revise it in another PR.

Reply:
i didn't get this tbh :D @aromko

Reply:
If you look into this file _src/routes/__root.tsx_ you will find the  <TanStackRouterDevtools />. I would place the <ReactQueryDevtools />next to it.